File: captainhcg/go/167-two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
    left, right := 0, len(numbers) - 1
    for left < right{
        v := numbers[left] + numbers[right]
        if v == target{
            return []int{left+1, right+1}
        } else if v < target{
            left += 1
        } else{
            right -= 1
        }
    }
    return []int{-1, -1}
}

File: bwu/array/167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        n = len(numbers)
        if n < 2:
            return []
        head, tail = 0, n - 1
        while head < tail:
            tmp = numbers[head] + numbers[tail]
            if tmp == target:
                return [head + 1, tail + 1]
            elif tmp < target:
                head += 1
            else:
                tail -= 1
                
        return []
